### ISSUE TITLE
Treat intra-doc links as references to be renamed or found

### ISSUE THREAD

**Issue by kpreid** [rust-analyzer/rust-analyzer#8242]
It seems that intra-doc links are not counted as references for purposes of Rename or Find All References. It would be useful if they were. (This might count as part of #6168, but the description there seems focused on following links rather than finding them.)

The support for this would also probably make it easy to show broken links as problems immediately (right now, I have to run `cargo doc` to check for them).

**Comment by Veykril** [comment]
This definitely sounds like something desirable but requires a bit of design in how we do the search for them, as references in docs aren't proper nodes like other names but just parts of the comment nodes.

**Comment by matklad** [comment]
> requires a bit of design

Yeah, we need ide::Refernece, like we have ide::Definition, to handle this clearly.

**Comment by clarfonthey** [comment]
I'm not sure what exactly needs to be done to get this to work. Right now, if I rename a symbol, I have to update all of the intradoc links manually, since they aren't touched.

One potentially weird case for this is that imports which are marked as `#[cfg(doc)]` may apply when resolving these links.
